pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -0,0 +1,625 @@
+    optional int32 host_port = 4;
+    // The host IP.
+    optional string host_ip = 5;
+}
+
+// Mount specifies the volume mount for the sandbox
+message Mount {
+    // The name of the volume mount.
+    optional string name = 1;
+    // The path of the mount within the container.
+    optional string container_path = 2;
+    // The path of the mount on the host.
+    optional string host_path = 3;
+    // If set, the mount is read-only.
+    optional bool readonly = 4;
+    // If set, the mount needs SELinux relabeling
+    optional bool selinux_relabel = 5;
+}
+
+// ResourceRequirements contains a set of resources
+// Valid resources are:
+// - cpu, in cores. (500m = .5 cores)
+// - memory, in bytes. (500Gi = 500GiB = 500 * 1024 * 1024 * 1024)
+message ResourceRequirements {
+    // The maximum amount of compute resources allowed.
+    optional double limits = 1;
+    // The minimum amount of compute resources required.
+    // If Request is omitted for a container, it defaults to Limits if that is explicitly specified, otherwise to an implementation-defined value
+    optional double requests = 2;
+}
+
+// PodSandboxResources contains the CPU/memory resource requirements.
+message PodSandboxResources {
+    // CPU resource requirement.
+    optional ResourceRequirements cpu = 1;
+    // Memory resource requirement.
+    optional ResourceRequirements memory = 2;
+}
+
+// NamespaceOption provides options for Linux namespaces.
+message NamespaceOption {
+    // If set, use the host's network namespace.
+    optional bool host_network = 1;
+    // If set, use the host's pid namesapce.
+    optional bool host_pid = 2;
+    // If set, use the host's ipc namespace.
+    optional bool host_ipc = 3;
+}
+
+// LinuxPodSandboxConfig holds platform-specific configuraions for Linux
+// host platforms and Linux-based containers.
+message LinuxPodSandboxConfig {
+    // The parent cgroup of the pod sandbox.
+    // The cgroupfs style syntax will be used, but the container runtime can 
+    // convert it to systemd semantices if needed.
+    optional string cgroup_parent = 1;
+    // The configurations for the sandbox's namespaces.
+    // This will be used only if the PodSandbox uses namespace for isolation.
+    optional NamespaceOption namespace_options = 2;
+}
+
+message CreatePodSandboxRequest {
+    // The configuration for creating a PodSandBox.
+    optional PodSandboxConfig config = 1;
+}
+
+message CreatePodSandboxResponse {
+    // The id of the PodSandBox
+    optional string pod_sandbox_id = 1;
+}
+
+message DeletePodSandboxRequest {
+    // The id of the PodSandBox
+    optional string pod_sandbox_id = 1;
+}
+
+message DeletePodSandboxResponse {}
+
+message PodSandboxStatusRequest {
+    // The id of the PodSandBox
+    optional string pod_sandbox_id = 1;
+}
+
+// PodSandboxNetworkStatus is the status of the network for a PodSandbox.
+message PodSandboxNetworkStatus {
+    // The IP address of the PodSandbox
+    optional string ip = 1;
```

Comment:
@yifan-gu Please open another issue to discuss supporting more network feautures, e.g. multiple interfaces and ipv6.